_studio/mfx_lib/vpp/include/mfx_vpp_hw.h

```diff
@@ -991,7 +991,7 @@ namespace MfxHwVideoProcessing
 
         CmCopyWrapper *m_pCmCopy;
 
-#if defined(MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP)
+#if defined(MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP) && defined(MFX_ENABLE_ASC)
```

Comment:
Are there any difference between MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP and MFX_ENABLE_ASC? @onabiull, @alariono : do you know?

Reply:
This question still remains. @onabiull, @alariono : your comments?

Reply:
@dvrogozh Look like this is very long history, ask @spossos for details 
asc_genx_bdw kernel - SCD component -> file mfx_scd.cpp
genx_scd_bdw kernel - ASC component-> file asc.cpp

So, for FB we need to disable both...

Reply:
I believe ASC is "advanced mode" of SCD, but regular mode looks deprecated a long ago. Still not sure if we have non-advanced mode functional after this change.

Reply:
The last element that was using non advanced SCD was ADI, but it has been updated to ASC. Regular SCD needs to be removed.

Reply:
@spossos : as far as this patch is concerned, would that be correct to:
1. Substitute MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP with MFX_ENABLE_ASC everywhere this patch has touched
2. Add the following somewhere in config.h:
```
#if defined(MFX_ENABLE_ASC)
  #define MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP
#endif
```

Some other patches will need to cleanup the rest of MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP places?

Reply:
@dvrogozh,  @spossos: what will be the final decision?
1. Substitute MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP with MFX_ENABLE_ASC 
2. Second patch with substitution MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP with MFX_ENABLE_ASC in the rest of code?

Reply:
That change looks safe, I analyzed the code and I agree with it.

Reply:
We have discussed the topic with @spossos and we have few important changes to make. Here is a summary:
1. In open source code old implementation for scene change detection is completely wiped out and substituted by ASC. Secondly, ASC is used not only in VPP but also in AVC encoder. I suggest the following decision:
1.1. MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP should remain in VPP to cover the named feature (do not add MFX_ENABLE_ASC)
1.2. The following code to be added in the mfx_config.h:
```
#if defined(MFX_ENABLE_ASC)
    #define MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP
#endif
```
2. We should wrap with the define (either MFX_ENABLE_ASC or better MFX_AVC_ENABLE_SOME_FEATURE) the code in AVC encoder which depends on ASC. We did not see the build failure because our fastboot don't enable AVC encoder. I believe this can be done later on as a follow up after the patch. @spossos : will you take the task?
3. MCTF actually depends on ASC and can't work without it.

Alternative decision for 1) and 2) could be to use MFX_ENABLE_ASC all over the VPP and AVC encoder. This would be that MFX_ENABLE_SCENE_CHANGE_DETECTION_VPP will be removed, and we will loss an opportunity to enable/disable these features in VPP/AVC separately. I am not sure that we will need this though. @onabiull : how do you think?

Reply:
While it would require more work, I believe first proposed approach would be better. Thus we'll reduce unexpected surprises.